### ISSUE TITLE
ParseException when using a lambda-expression as a parameter in an enum

### ISSUE THREAD

**Issue by RobertKolter** [paul-hammant/qdox#50]
Using a lambda expression as a parameter in an enum doesn't work in version 2.0-M9. 

The following class is passed to JavaprojectBuilder.addSource

```java
... // some Imports...
public class ClassToTest {
	public static enum Display {
		LOWVALUE(v -> v.toLowerCase() + "_");

		final Function<String, String> convert;

		private Display(final Function<String, String> convert) {
			this.convert = convert;
		}
	}
}
``` 

This gives
```
Exception in thread "main" com.thoughtworks.qdox.parser.ParseException: syntax error @[8,15] in file:[...]
	at com.thoughtworks.qdox.parser.impl.Parser.yyerror(Parser.java:1775)
	at com.thoughtworks.qdox.parser.impl.Parser.yyparse(Parser.java:1897)
	at com.thoughtworks.qdox.parser.impl.Parser.parse(Parser.java:1756)
	at com.thoughtworks.qdox.library.SourceLibrary.parse(SourceLibrary.java:232)
	at com.thoughtworks.qdox.library.SourceLibrary.parse(SourceLibrary.java:209)
	at com.thoughtworks.qdox.library.SourceLibrary.addSource(SourceLibrary.java:159)
	at com.thoughtworks.qdox.library.SortedClassLibraryBuilder.addSource(SortedClassLibraryBuilder.java:174)
	at com.thoughtworks.qdox.JavaProjectBuilder.addSource(JavaProjectBuilder.java:151)
```

Is this bug already fixed in some version for the near future? Thanks in advance.

**Comment by RobertKolter** [comment]
As you have written in https://github.com/paul-hammant/qdox/issues/50 I have checked the grammar file. I found that the following part of the grammar for "Block" seems to differ from the JLS. 

The JLS and the grammer of jdox speifies the following:
```
LambdaBody:
   Expression
   Block 
```

While the JLS then specifies 
```
Block:
   { [BlockStatements] }

BlockStatements:
   BlockStatement {BlockStatement}

BlockStatement:
   LocalVariableDeclarationStatement
   ClassDeclaration
   Statement
```
the grammar of jdox doesn't seem to provide a description for "Block". maybe this could explain the behaviour.
